### ISSUE TITLE
#[allow(clippy::all)] doesn't cover clippy::pedantic

### ISSUE THREAD

**Issue by hellow554** [rust-lang/rust-clippy#8750]
### Summary

When you want to generate code via `bindgen`, you typically have something like this in your `lib.rs`

```rust
mod bindings {
    include!(concat!(env!("OUT_DIR"), "/bindgen.rs"));
}
```

To void all sorts of compiler warnings you add `#[allow(...)]` on that `bindings` mod.

Problem is, although `clippy::all` is specified, lints from the pedantic category don't fall into that group.

### Reproducer

I tried this code ([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8532daee982f0ad800a54b8320eb3495)):  

```rust
#![warn(clippy::pedantic)]

#[allow(clippy::all)]
pub mod ffi {
    pub const X: f64 = 3.14; // approx_constant is suppressed
    pub const NUM: u32 = 129381923; // unreadable_literal is not suppressed
}

```

I expected to see this happen: no warnings

Instead, this happened:

```
warning: long literal lacking separators
 [--> src/lib.rs:6:26
](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8532daee982f0ad800a54b8320eb3495#)  |
6 |     pub const NUM: u32 = 129381923; // unreadable_literal is not suppressed
  |                          ^^^^^^^^^ help: consider: `129_381_923`
  |
note: the lint level is defined here
 [--> src/lib.rs:1:9
](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8532daee982f0ad800a54b8320eb3495#)  |
1 | #![warn(clippy::pedantic)]
  |         ^^^^^^^^^^^^^^^^
  = note: `#[warn(clippy::unreadable_literal)]` implied by `#[warn(clippy::pedantic)]`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unreadable_literal
```

**Comment by mikerite** [comment]
I asked on Zulip what generated code should do to silence all lints a while ago - [link](https://rust-lang.zulipchat.com/#narrow/stream/257328-clippy/topic/Allows.20in.20generated.20code/near/246552338). The answer I got was `#![allow(clippy::all, clippy::pedantic, clippy::restriction, clippy::nursery)]` .

**Comment by hellow554** [comment]
Thanks for the link! :) 

I'm not sure if that's a good way though. `clippy::all` doesn't cover all clippy lints seems odd to me.

**Comment by matthiaskrgr** [comment]
I think "all" just means the "default" lints, and everything that you enable "manually" also will have to be silenced manually. 

`clippy::default` might be a more appropriate lint group name but I would not want to deal with the fallout of renaming it... 🙃
